### ISSUE TITLE
Nextcloud 10, internal XMPP Server, LDAP

### ISSUE THREAD

**Issue by joergmschulz** [jsxc/jsxc#374]
For testing, I use the internal JSXC XMPP server
While I get the chat functionality to work well for non-LDAP users in NC10, LDAP users can only send messages while they receive none.
Sometimes, these LDAP users are listed as unknown users in the context of the recipient when the LDAP user sends a message.

Where can I access logs of the internal server to further investigate the issue?

**Comment by MarcelWaldvogel** [comment]
We will have a look at this.

However, if you use LDAP, you are probably big enough to run a real XMPP server, which also has the option to interoperate with the XMPP federation.

**Comment by joergmschulz** [comment]
Sure, that's the plan  - it's in test on a ProofOfConcept server anyway. If you point me to a place where I can see/activate some logging, I am willing to step in.

**Comment by MarcelWaldvogel** [comment]
@LEDfan, can you help?

**Comment by LEDfan** [comment]
I'll have a look a this. @joergmschulz can you provide the following information, that will make it easier to debug:
 - ownCloud or Nextcloud version 
 - post the owncloud.log or nextcloud.log (in the data directory)
 - post the browser console log

Thanks

**Comment by joergmschulz** [comment]
So, we have NC Nextcloud 10.0.1 (beta) and the Nextcloud 10.0.1 likewise. The issue is reproducible in both versions.
The browser console log is empty.
The (beta) version has quite verbose logging. I'll try to summarize the results because I see lots of LDAP messages that make sense and no error messages.
First of all, a set of screenshots:
LDAP User joerg.schulz@elternserver.de
Everything looks fine, especially user js is marked as active; we have a chat window with messages to and from js. But look at the 2nd screenshot...
![ldapuser](https://cloud.githubusercontent.com/assets/6523646/19006393/8dfd3188-875f-11e6-99ee-eaa876d06e2a.png)
Here, we have the NON-LDAP user js
![nonldapuser](https://cloud.githubusercontent.com/assets/6523646/19006473/ebae4dbc-875f-11e6-859e-3a0d15c84325.png)
the LDAP user is NOT displayed as active.
But, chats could be received and sent.
The left chat window displays the correct credentials. I opened this window by clicking on the target LDAP username. These chats are sent to the LDAP user, correctly. They arrive.
The right chat window displays an 'unknown user' with somehow correct albeit incomplete credentials. I opened the window after the first chat arrived (signal: small orange dot beside the online mark). Here, chats are received. But the chats entered here never arrive anywhere.
Now it gets interesting - the LDAP queries.
When I send chats FROM the LDAP user, there are queries, these chats arrive in the right window at user js.
When I enter answers from that window, no LDAP queries are submitted.
When I enter answers from the left window, ldap queries start and the chat arrives.

So, obviously, the app for the non-ldap user cannot associate the chats from the ldap user to the correct entry in the database because of ???? It doesn't even do ldap lookups.
